scripts/incremental_validator.py

```
@@ -0,0 +1,322 @@
+        except Exception as e:
+            print(f"Error saving validation cache: {e}")
+            
+    def calculate_file_hash(self, file_path: Path) -> str:
+        """Calculate SHA256 hash of a file."""
+        hash_sha256 = hashlib.sha256()
+        try:
+            with open(file_path, "rb") as f:
+                for chunk in iter(lambda: f.read(4096), b""):
+                    hash_sha256.update(chunk)
+            return hash_sha256.hexdigest()
+        except Exception:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)